tests/test/org/commcare/cases/test/CaseParseAndReadTest.java

```diff
@@ -0,0 +1,79 @@
+package org.commcare.cases.test;
+
+import org.commcare.test.utils.TestInstanceInitializer;
+import org.commcare.test.utils.XmlComparator;
+import org.commcare.util.mocks.MockDataUtils;
+import org.commcare.util.mocks.MockUserDataSandbox;
+import org.javarosa.core.io.StreamsUtil;
+import org.javarosa.core.model.instance.ExternalDataInstance;
+import org.javarosa.model.xform.DataModelSerializer;
+import org.junit.Before;
+import org.junit.Test;
+import org.kxml2.kdom.Document;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Test suite to verify end-to-end parsing of inbound case XML
+ * and reading values back from the casedb model
+ *
+ * @author ctsims
+ */
+public class CaseParseAndReadTest {
+
+    private MockUserDataSandbox sandbox;
+
+    @Before
+    public void setUp() {
+        sandbox = MockDataUtils.getStaticStorage();
+    }
+
+    @Test
+    public void testReadCaseDB() throws IOException {
+        compareCaseDbState("/case_create_basic.xml", "/case_create_basic_output.xml");
+    }
+
+    private void compareCaseDbState(String inputTransactions, String caseDbState) throws IOException {
+        MockDataUtils.parseIntoSandbox(this.getClass().getResourceAsStream(inputTransactions), sandbox);
+
+        byte[] parsedDb = dumpInstance("jr://instance/casedb");
+        Document parsed = XmlComparator.getDocumentFromStream(new ByteArrayInputStream(parsedDb));
+        Document loaded = XmlComparator.getDocumentFromStream(this.getClass().getResourceAsStream(caseDbState));
+
+        try {
+            XmlComparator.compareXmlDOMs(parsed, loaded);
+        } catch (Exception e) {
+            System.out.print(new String(parsedDb));
+
+
+            assertEquals("CaseDB output did not match expected structure(" + e.getMessage() + ")", new String(dumpStream(caseDbState)), new String(parsedDb));
```

Comment:
Hey,

So this is actually basically an "assertFalse" that happens to outline the correct sections of the xml more often than not.

The two strings aren't actually guaranteed to be the same even _if_ the XML is semantically equivalent, but if they _are_ different, the strings definitely will be, and the areas where they are different get highlighted quite nicely in my tests